common/test/acceptance/pages/studio/settings_graders.py

```diff
@@ -11,6 +14,261 @@ class GradingPage(CoursePage):
+        """
+        Set letter grade name by index.
+        """
+        index = self._get_index_of('.letter-grade', letter)
+        self.q(css='.letter-grade').nth(index).fill(value)
+
+    def move_grading_range_by_offset(self, letter, offset_x=0, offset_y=0):
+        """
+        Moves grading range by the offset.
+        """
+        # First element is not removable, that's why we should subtract 1
+        index = self._get_index_of('.letter-grade', letter) - 1
+        draggable = self.q(css='.grade-range .ui-resizable-e').results[index]
+        action = ActionChains(self.browser)
+        action.drag_and_drop_by_offset(draggable, offset_x, offset_y).release().perform()
+        self.click_save_button()
+
+    def refresh(self):
+        """
+        Reloads the page.
+        """
+        self.browser.refresh()
+        self.wait_for_page()
+
+    def wait_for_assignmnents(self):
+        """
+        Ensure the assignments available for use.
+        """
+        EmptyPromise(
+            lambda: self.q(css='.course-grading-assignment-list-item').present,
+            'Assignment types are finished rendering'
+        ).fulfill()
+
+    def wait_for_letter_grades(self):
+        """
+        Ensure the grading ranges are available for use.
+        """
+        EmptyPromise(
+            lambda: len(self.letter_grades) > 0,
+            'Gradeing ranges are rendered.'
+        ).fulfill()
+
+    def wait_for_grace_period(self, value):
+        """
+        Ensure the grading ranges are available for use.
+        """
+        EmptyPromise(lambda: self.grace_period == value, 'Grace period is changed.').fulfill()
+
+    def wait_for_confirmation_prompt(self):
+        """
+        Show confirmation prompt.
+
+        We can't use confirm_prompt because its wait_for_notification
+        is flaky when asynchronous operation completed very quickly.
+        """
+        self.wait_for_element_visibility('.notification', 'Notofication is visible')
+        self.wait_for_element_visibility('.notification .action-primary', 'Confirmation button is visible')
+
+    def is_confirmation_message_visible(self):
+        """
+        Indicates whether confirmation message visible.
+        """
+        return self.q(css="#alert-confirmation").visible
+
+    def is_save_button_disabled(self):
+        """
+        Indicates whether save button is disabled.
+        """
+        return self.q(css="div#page-notification button.action-save.is-disabled").present
+
+    def click_add_assignment_button(self):
+        """
+        Clicks the 'New Assignment Type' button.
+        """
+        self.q(css=self.grading_css + " .add-grading-data").first.click()
+
+    def click_enable_passing_grade(self):
+        """
+        Clicks the 'Passing Grade Enabled' checkbox.
+        """
+        qs = self.q(css=self.grading_css + " #assignment-passing-grade-enabled")
+        qs[0].click()
+        qs[-1].click()
```

Comment:
> Why do we need 2 clicks?

Because this is one possible way to enable last Assignment `passing_grade_enable` checkbox.
May it is an issue due to repeated id's on page.